src/libs/ValidationUtils.js

```diff
@@ -86,58 +85,44 @@ function isValidURL(url) {
 
 /**
  * @param {Object} identity
- * @returns {Boolean}
+ * @returns {Object}
  */
-function isValidIdentity(identity) {
+function validateIdentity(identity) {
+    const errors = {};
     if (!isValidAddress(identity.street)) {
-        showBankAccountFormValidationError(translateLocal('bankAccount.error.address'));
-        showBankAccountErrorModal();
-        return false;
+        errors.street = true;
     }
 
-    if (identity.state === '') {
-        showBankAccountFormValidationError(translateLocal('bankAccount.error.addressState'));
-        showBankAccountErrorModal();
-        return false;
+    if (!isRequiredFulfilled(identity.state)) {
+        errors.state = true;
     }
 
-    if (identity.city === '') {
-        showBankAccountFormValidationError(translateLocal('bankAccount.error.addressCity'));
-        showBankAccountErrorModal();
-        return false;
+    if (!isRequiredFulfilled(identity.city)) {
+        errors.city = true;
     }
 
     if (!isValidZipCode(identity.zipCode)) {
-        showBankAccountFormValidationError(translateLocal('bankAccount.error.zipCode'));
-        showBankAccountErrorModal();
-        return false;
+        errors.zipCode = true;
     }
 
     if (!isValidDate(identity.dob)) {
-        showBankAccountFormValidationError(translateLocal('bankAccount.error.dob'));
-        showBankAccountErrorModal();
-        return false;
-    }
-
-    if (!isValidAge(identity.dob)) {
-        showBankAccountFormValidationError(translateLocal('bankAccount.error.age'));
-        showBankAccountErrorModal();
-        return false;
+        errors.dob = true;
+    } else if (!isValidAge(identity.dob)) {
```

Comment:
There is an `else if` because I didn't want to try to validate the age if the date is already invalid. This mattered when I was storing both errors as text using the same key `dob`, but maybe it doesn't matter anymore. I'll have a look tomorrow.

Reply:
I think there hasn't been a conversation about that, but I would think it should be just one error, and in a case like this, we are giving priority to "invalid date"

Reply:
About this, I lean toward leaving it like it is (else if) because I prefer not to validate an "age" of an invalid date.

Reply:
Should we just do...

>Only show the invalid age error if we have a valid date

Sounds like we might want to change this in the future. But I think for the purposes of this PR it's not something we should get stuck on.